src/app.py

```diff
@@ -52,7 +88,85 @@ async def add_headers(request: Request, call_next: Callable) -> Response:
     return response
 
 
-@app.get("/api/download", tags=["stable"])
+class RoleChecker:
+    def __init__(self, allowed_roles: List[str]) -> None:
+        self.allowed_roles = allowed_roles
+
+    def __call__(self, user: User = Depends(current_user)):
+        if not any(role in self.allowed_roles for role in user.roles):
+            raise HTTPException(
+                status_code=403,
+                detail=f"Operation not allowed for user {user.name} (roles: {user.roles})",
```

Comment:
Maybe add information what roles are required?

Reply:
Good point, added. Originally I was not sure if that has any security implications so I've stayed on the safe side.